File: surveilance-UI/src/types/employee.ts
export interface Employee {
    id: string;
    locationId?: string | null;
    firstName: string;
    lastName: string;
    email: string;
    employeeId: string;
    number?: string;
    companyName?: string;
    designation?: string;
    department?: string;
    tenure?: string;
    grade?: string;
    gender?: string;
    managerId?: string;
    metadata?: Record<string, any>;
    faceScanStatus: 'NotAssigned' | 'Pending' | 'Completed';
    faceScanCompletedAt?: string;
    faceScanInviteSentAt?: string;
    createdAt: string;
    updatedAt: string;
}

export interface EmployeeRequest {
    firstName: string;
    lastName: string;
    email: string;
    employeeId: string;
    /**
     * Pass a GUID to assign / change Location.
     * Pass '00000000-0000-0000-0000-000000000000' to detach.
     * Omit / null leaves it unchanged on update.
     */
    locationId?: string | null;
    number?: string;
    companyName?: string;
    designation?: string;
    department?: string;
    tenure?: string;
    grade?: string;
    gender?: string;
    managerId?: string;
    metadata?: Record<string, any>;
}

export interface BulkImportResponse {
    totalProcessed: number;
    successCount: number;
    failureCount: number;
    failures: BulkImportFailure[];
}

export interface BulkImportFailure {
    rowIndex: number;
    email: string;
    reason: string;
}

File: violation-management-service-api/violation-management-api/Core/Domain/Employee.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using AlphaSurveilance.Core.Enums;

namespace AlphaSurveilance.Core.Domain
{
    public class Employee
    {
        [Key]
        public Guid Id { get; set; } = Guid.NewGuid();

        [Required]
        public string TenantId { get; set; } = string.Empty;

        [Required]
        public string FirstName { get; set; } = string.Empty;

        [Required]
        public string LastName { get; set; } = string.Empty;

        [Required]
        public string Email { get; set; } = string.Empty;

        [Required]
        public string EmployeeId { get; set; } = string.Empty;

        public string? Number { get; set; }

        public string? CompanyName { get; set; }

        public string? Designation { get; set; }

        public string? Department { get; set; }

        public string? Tenure { get; set; }

        public string? Grade { get; set; }

        public string? Gender { get; set; }

        public string? ManagerId { get; set; }

        // Stores Skills, Certifications, Languages, and other extra fields as JSON
        [Column(TypeName = "jsonb")]
        public string? MetadataJson { get; set; }

        public FaceScanStatus FaceScanStatus { get; set; } = FaceScanStatus.NotAssigned;
        public DateTime? FaceScanCompletedAt { get; set; }
        public DateTime? FaceScanInviteSentAt { get; set; }

        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
        public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;
    }
}
